modules/ipfs-cpinner-provider/src/IpfsPinnerProvider.ts

```diff
@@ -20,10 +20,10 @@ export default class IpfsPinnerProvider {
     return cid
   }
 
-  async get (did: DID, key: Key): Promise<Content[]> {
+  async get (did: DID, key: Key): Promise<SavedContent[]> {
     const cids = await this.metadataManager.find(did, key)
 
-    return Promise.all(cids.map(cid => this.ipfsClient.get(cid))).then(arrays => Array.prototype.concat.apply([], arrays))
+    return Promise.all(cids.map(cid => this.ipfsClient.get(cid).then(([content]) => ({ id: cid, content }))))
```

Comment:
Improved legibility:

```suggestion
    return Promise.all(
      cids.map(cid => this.ipfsClient.get(cid)
        .then(([content]) => ({ id: cid, content }))
      )
    )
```

Reply:
I'm now reading `ipfsClient` implementation again and something seems weird. We are getting many files iterating but then using just the first one